draft-lmbdhk-quic-multipath.md

```diff
@@ -517,45 +518,111 @@ not cause linkability issue.
+zero-length CID, the client needs to use path identifier type 0x01 with a
+path_id=2 in its PATH_ABANDON frame, which is different from the above example.
+The server can readily close the first path when it receives the
+RETIRE_CONNECTION_ID frame from the client. However, since the client will not
+receive a RETIRE_CONNECTION_ID frame, after sending out RETIRE_CONNECTION_ID
+frame, the client waits for 3 RTO before closing the path. Note that single
+packet number spaces are used in both directions.
 
-In scenarios such as client detects the network environment change (client's 4G/Wi-Fi is turned off,
-Wi-Fi signal is fading to a threshold), or endpoints detect that the quality of RTT or loss rate is
-becoming worse, client or server can terminate a path immediately.
+~~~
+  Client                                                      Server
+
+  (client tells server to abandon a path)
+  1-RTT[X]: DCID=S2 PATH_ABANDON[path_id_type=1, path_id=2]->
+                             (server tells client to abandon a path)
+                <-1-RTT[Y]: PATH_ABANDON[path_id_type=0, path_id=2],
+                                                           ACK[PN=X]
+  (client retires the CID it is using
+  and waits for 3*RTO to close the path)
+  1-RTT[X+1]: DCID=S3 RETIRE_CONNECTION_ID[2],
+                                     ACK[PN=Y]->
+                                            (server closes the path)
+
+~~~
+{: #fig-example-path-close2 title="Example of closing a path when the client
+uses non-zero length CID while the server uses zero-length CID"}
+
+{{fig-example-path-close3}} illustrates an example of path closing when both
+the client and the server use zero-length CIDs. Again, we are going to close
+the first path. In this case, both the server and the client use path
+identifier type 0x02, which refers to the current path in the PATH_ABANDON
+frame. Both endpoints wait for 3 RTO after receiving the ACK of their
+respective PATH_ABANDON frames before closing the path. Note that single packet
+number spaces are used in both directions.
+
+~~~
+  Client                                                     Server
+
+  (client tells server to abandon a path)
+  1-RTT[X]: DCID=S2 PATH_ABANDON[path_id_type=2]->
+                            (server tells client to abandon a path)
+                          <-1-RTT[Y]: PATH_ABANDON[path_id_type=2],
+                                                          ACK[PN=X]
+  (client waits for 3*RTO to close the path)
+  1-RTT[X+1]: ACK[PN=Y]->
+                         (server waits for 3*RTO to close the path)
+
+~~~
+{: #fig-example-path-close3 title="Example of closing a path when both client
+and server use zero-length CIDs."}
+
+In scenarios such as client detects the network environment change (client's
+4G/Wi-Fi is turned off, Wi-Fi signal is fading to a threshold), or endpoints
+detect that the quality of RTT or loss rate is becoming worse, client or server
+can terminate a path immediately.
```

Comment:
I suppose it can terminate using a path immediately and maybe send a PATH_ABANDON later, but I assume the use case here is to say that PATH_ABANDON is a kind of graceful path closure, but abrupt closures are also possible.